src/components/List/List.js

```diff
@@ -1,40 +1,52 @@
 import React from 'react';
 
-import ReactDragList from 'react-drag-list';
-import 'react-drag-list/assets/index.css';
-
 import './List.css';
 
 class List extends React.Component {
-  _handleUpdate = (evt, updated) => {
-    console.log(evt); // tslint:disable-line
-    console.log(updated); // tslint:disable-line
-    // this.setState({
-    //   dataSource: [...updated, {
-    //     color: '#FFAA00',
-    //     title: 'Added Engineer',
-    //     text: 'Added Engineer',
-    //   }]
-    // })
-  };
+  constructor(props) {
+    super(props);
+
+    this.sortPlayers = this.sortPlayers.bind(this);
+    this.handleInitiativeChange = this.handleInitiativeChange.bind(this);
+  }
+
+  handleInitiativeChange(event, player) {
+    let initiative = event.target.value;
+
+    this.props.handleInitiativeChange(initiative, player);
+    event.preventDefault();
+  }
+
+  sortPlayers() {
+    // sort by value
+    let initiativeOrderArray = this.props.players.sort(function(a, b) {
```

Comment:
The `Array.prototype.sort` method is a little tricky since it sorts the array in-place, and so would be trying to modify the props given to this component here! 🙅‍♂️ This might be where you were running into some weird behavior with sorting perhaps... instead, I'd make a copy of the array and sort that:

```js
let initiativeOrderArray = this.props.players.concat().sort(function(a, b) {
  return a.initiative - b.initiative;
});

```

Reply:
This could also be a nice place to use Lodash's [`sortBy`](https://lodash.com/docs/4.17.10#sortBy):

```js
import { sortBy } from 'lodash';

let initiativeOrderArray = sortBy(this.props.players, 'initiative');
```